exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/DrillFileSystem.java

```diff
@@ -89,22 +89,36 @@ public DrillFileSystem(Configuration fsConf) throws IOException {
   }
 
   public DrillFileSystem(Configuration fsConf, OperatorStats operatorStats) throws IOException {
-    this.underlyingFs = FileSystem.get(fsConf);
+    this(fsConf, URI.create(fsConf.getRaw(FS_DEFAULT_NAME_KEY)), operatorStats);
+  }
+
+  public DrillFileSystem(Configuration fsConf, URI Uri, OperatorStats operatorStats) throws IOException {
+    this.underlyingFs = FileSystem.get(Uri, fsConf);
+    logger.trace("Configuration for the DrillFileSystem " + fsConf.getRaw(FS_DEFAULT_NAME_KEY) +
+        ", underlyingFs: " + this.underlyingFs.getUri());
     this.codecFactory = new CompressionCodecFactory(fsConf);
     this.operatorStats = operatorStats;
+    setConf(fsConf);
   }
 
   @Override
   public void setConf(Configuration conf) {
```

Comment:
This is existing code. But, the existence of this method is probably very poor design. This method says that we create a file system using one config, then change the config later. Does the class work if we start with, say, "file:///" then change to HDFS? The `setConf()` method should probably be removed and code in this class should say that the config is immutable once the `DrillFileSystem` is created. Else, all bets are off on semantics.

Reply:
Agreed. Unfortunately setConf() is a public method declared in hadoop's FileSystem class. Additionally setConf() is called with a null argument everytime DrillFileSystem is initialized because the FileSystem constructor calls the Configured(Configuration conf) constructor and passes null for the configuration.

What I can do is make the semantics of DrillFileSystem explicit by documenting that DrillFileSystem.setConf should never be called and by throwing an exception if DrillFileSystem.setConf is ever called after a DrillFileSystem object is constructed.